src/test/jtx/token.h

```diff
@@ -0,0 +1,231 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2021 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_TEST_JTX_NFT_H_INCLUDED
+#define RIPPLE_TEST_JTX_NFT_H_INCLUDED
+
+#include <test/jtx/Account.h>
+#include <test/jtx/Env.h>
+#include <test/jtx/owners.h>
+
+#include <ripple/basics/strHex.h>
+
+#include <initializer_list>
+
+namespace ripple {
+namespace test {
+namespace jtx {
+
+namespace token {
+
+/** Mint an NFToken. */
+Json::Value
+mint(jtx::Account const& account, std::uint32_t tokenTaxon = 0);
+
+/** Sets the optional TransferFee on an NFTokenMint. */
+class xferFee
+{
+private:
+    std::uint16_t xferFee_;
+
+public:
+    xferFee(std::uint16_t fee) : xferFee_(fee)
+    {
+    }
+
+    void
+    operator()(Env&, JTx& jtx) const;
+};
+
+/** Sets the optional Issuer on an NFTokenMint. */
+class issuer
+{
+private:
+    std::string issuer_;
```

Comment:
I see we're storing string representation of the wrapped object rather then the object itself (here and in the other classes). I'd probably store the object - they are 20-byte object, there's not much cost to copying them and storing them. But it's also fine as-is if you prefer it.

Reply:
Storing it as a string makes it a little easier to use, since we're just about to use it to build JSON.  I'll leave as is.  But thanks for the comment.

Reply:
Storing it as a string makes it a little easier to use, since we're just about to use it to build JSON.  I'll leave as is.  But thanks for the comment.